pkg/oci/load_balancer_spec.go

```diff
@@ -174,34 +174,32 @@ func (s *LBSpec) Certificates() (map[string]loadbalancer.CertificateDetails, err
 		return certs, nil
 	}
 	//Read listener Kubernetes Secret
-	sslSecret, err := s.SSLConfig.readSSLSecret(s.service.Namespace, s.SSLConfig.ListenerSSLSecretName)
-	if err != nil {
-		return nil, errors.Wrap(err, "reading SSL Listener Secret")
-	}
-	if len(sslSecret.PublicCert) == 0 || len(sslSecret.PrivateKey) == 0 {
-		return certs, nil
-	}
+	if len(s.SSLConfig.ListenerSSLSecretName) != 0 {
```

Comment:
We can support ca certificates and passphrases on listener certs and implement as follows I believe:

```go
	secrets := make([]string, 0, 2)
	if s.SSLConfig.ListenerSSLSecretName != "" {
		secrets = append(secrets, s.SSLConfig.ListenerSSLSecretName)
	}
	if s.SSLConfig.BackendSetSSLSecretName != "" {
		secrets = append(secrets, s.SSLConfig.BackendSetSSLSecretName)
	}

	for _, name := range secrets {
		cert, err := s.SSLConfig.readSSLSecret(s.service.Namespace, name)
		if err != nil {
			return nil, errors.Wrap(err, "reading SSL BackendSet Secret")
		}

		certs[name] = loadbalancer.CertificateDetails{
			CertificateName:   &s.SSLConfig.BackendSetSSLSecretName,
			CaCertificate:     common.String(string(cert.CACert)),
			PublicCertificate: common.String(string(cert.PublicCert)),
			PrivateKey:        common.String(string(cert.PrivateKey)),
			Passphrase:        common.String(string(cert.Passphrase)),
		}
	}